assets/Shaders/default.vert

```diff
@@ -0,0 +1,84 @@
+﻿#version 130
+
+struct Light
+{
+    vec3 position;
+    vec4 ambient;
+    vec4 diffuse;
+    vec4 specular;
+};
+
+struct MaterialColor
+{
+    vec4 ambient;
+    vec4 diffuse;
+    vec4 specular;
+	vec4 emission;
+    float shininess;
+}; 
+
+in vec3 iPosition;
+in vec3 iNormal;
+in vec2 iUv;
+in vec4 iColor;
+
+uniform mat4 uCurrentTranslateMatrix;
+uniform mat4 uCurrentScaleMatrix;
+uniform mat4 uCurrentRotateMatrix;
+uniform mat4 uCurrentTextureTranslateMatrix;
+uniform mat4 uCurrentProjectionMatrix;
+uniform mat4 uCurrentViewMatrix;
+
+uniform vec3 uEyePosition;
+uniform bool uIsLight;
+uniform Light uLight;
+uniform MaterialColor uMaterial;
+uniform bool uIsFog;
+uniform float uFogStart;
+uniform float uFogEnd;
+
+out vec2 oUv;
+out vec4 oColor;
+out float oFogFactor;
+
+void main()
+{
+	vec3 worldPos = (uCurrentTranslateMatrix * uCurrentRotateMatrix * uCurrentScaleMatrix * vec4(iPosition, 1.0)).xyz;
+	vec4 viewPos = uCurrentViewMatrix * vec4(worldPos, 1.0);
+	gl_Position = uCurrentProjectionMatrix * viewPos;
+
+	// Lighting
+	// Ambient
+	vec4 ambient  = uLight.ambient * uMaterial.ambient;
+
+	// Diffuse
+	vec3 norm = normalize(iNormal);
+	vec3 lightDir = normalize(uLight.position - worldPos);
+	float diff = max(dot(norm, lightDir), 0.0);
+	vec4 diffuse = uLight.diffuse * (diff * uMaterial.diffuse);
```

Comment:
```vec4 diffuse = (uLight.diffuse * (diff * uMaterial.diffuse)) * diff;```

I think that's where we're going wrong- That gives the expected results, although I need to test a bit more.

Reply:
This also seems to give approximately correct lighting, and thinking about is is probably the correct option:
```vec4 diffuse = (uLight.diffuse * diff) * (diff * uMaterial.diffuse);```